common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -170,7 +178,7 @@ protected void serviceInit(Configuration conf) throws Exception {
       List<TransportServerBootstrap> bootstraps = Lists.newArrayList();
       boolean authEnabled = conf.getBoolean(SPARK_AUTHENTICATE_KEY, DEFAULT_SPARK_AUTHENTICATE);
       if (authEnabled) {
-        createSecretManager();
+        createSecretManager(recoveryEnabled);
```

Comment:
I think at this point it would be cleaner to do:

```
secretManager = new ShuffleSecretManager();
if (recoveryEnabled) {
  loadSecretsFromDb();
}
```